static/js/lib/api/posts.js

```diff
@@ -47,11 +47,23 @@ export function updateRemoved(postId: string, removed: boolean): Promise<Post> {
   })
 }
 
-export function editPost(postId: string, post: Post): Promise<Post> {
-  return fetchJSONWithAuthFailure(`/api/v0/posts/${postId}/`, {
-    method: PATCH,
-    body:   JSON.stringify(R.dissoc("url", post))
-  })
+export async function editPost(postId: string, post: Post): Promise<Post> {
+  const formData = objectToFormData(R.pick(["text", "title"], post))
+
+  if (post.article_content && post.article_content.length !== 0) {
```

Comment:
Take or leave: `emptyOrNil` in `static/js/lib/util.js` could be used here and in `createPost`